release-notes.txt: add 2020.01 release notes

fjmolinas opened the pull request:
### Contribution description

Release notes for 2020.01. I've added the ~~[Don't merge yet]~~ since there is still two PR's waiting for a backport, so the statistics are not in yet.

Otherwise with regard to other releases I was much more extensive when listing all changes. If you think there is too much feel free to point out points that should be removed, or if a much more summarized version is required.

In some PR conversations I saw that @cladmi mentioned that by this release ` GNU make <4 is deprecated in RIOT`, I wasn't able to find a reference in code, does someone know where this is?

Other than that, if you are aware of deprecation or important removals I might have missed, please jump in!

### Testing procedure

Read the notes.

cladmi commented:
@fjmolinas https://github.com/RIOT-OS/RIOT/blob/f0cde06391f901a36bf959e352de494a8cd89971/Makefile.include#L1-L8

fjmolinas commented:
> @fjmolinas 

Thanks @cladmi !

basilfx commented on release-notes.txt:
```diff
@@ -1,3 +1,688 @@
+
+The 2020.01 release includes:
+
+ - Initial support for micropython
```
Change to MicroPython.

basilfx commented on release-notes.txt:
```diff
@@ -1,3 +1,688 @@
+ - Xtimer concurrency/robustness improvement, fixing #8388, #5338 & #5103
+ - Complete re-implementation of esp8266 based on ESP8266 RTOS SDK
+ - Improvements in automatic tests board coverage and test automation
+ - Introduce and start using kconfig as a configuration tool
```
Kconfig

basilfx commented on release-notes.txt:
```diff
@@ -1,3 +1,688 @@
+    * pkg/tinydtls: bump version
+    + pkg/tensorflow-lite: add support to RIOT
+    + pkg/utensor: add support to RIOT
+    * pkg/wakaama: Add basic LWM2M client implementation
```
Add: `* pkg/u8g2: refactor the U8g2 package`

(https://github.com/RIOT-OS/RIOT/commit/b253f61a7f3522b018fdbbd8d55f597fc06e61e7)

miri64 commented on release-notes.txt:
```diff
@@ -1,3 +1,688 @@
+
+ - Initial support for micropython
+ - Initial support for GNRC based LoRaWAN stack
+ - Initial asynchronous sock support
```
Maybe add an "experimental" here somewhere?

aabadie commented on release-notes.txt:
```diff
@@ -1,3 +1,688 @@
+    * core/ringbuffer: compensate 'ringbuffer_remove' underflow error
+    + core/thread: add zombie thread state
+    + sys: Basic C++ compatibility with C11 atomics
+    + sys/arduino: added Wire (I2C) interface
```
Would it be possible to somehow group addition/changes of the same area when there are several. Something like:
```
* sys/arduino:
  + added Wire (I2C) interface
  * allow use of arduino libraries...
  etc
```

?

aabadie commented on release-notes.txt:
```diff
@@ -1,3 +1,688 @@
+    + boards/derfmega*: add initial support
+    + boards/esp32-ttgo-t-beam: add initial support
+    - boards/frdm: remove support for OpenOCD < v0.10.0
+    * boards/hifive1b: configure ADC channels
```
There is no support for ADC on hifive1b.

fjmolinas replied:
Must have been a mix up with `boards/arduino-zero: configure ADC channels`

fjmolinas replied:
Thanks for the catch.

aabadie commented on release-notes.txt:
```suggestion
============
```

aabadie commented on release-notes.txt:
```suggestion
- GNU make <4 will be removed after release 2020.01
```

aabadie commented on release-notes.txt:
```suggestion
- gnrc_nettest: will be removed after release 2020.07
```

aabadie replied:
Maybe move this at the end of the list to keep a chronological order (closest from removal at the top)

miri64 commented on release-notes.txt:
```diff
@@ -1,3 +1,688 @@
+------
+    - boards: remove RTT_NUMOF/RTC_NUMOF
+    - boards: unexport PROGRAMMER
+    * boards/arduino-zero: configure ADC channels
```
Missing the adafruit-nrf52840 in the list.

fjmolinas replied:
Right, I missed tagging the milestone, so it slipped by, thanks!

aabadie commented on release-notes.txt:
```suggestion
- net/gcoap: gcoap_finish() will be removed after release 2020.04
```

aabadie commented on release-notes.txt:
```suggestion
- makefiles/vars.inc.mk: PORT_BSL & AVRDUDE_PORT will be removed after release 2020.04
```

aabadie commented on release-notes.txt:
```diff
@@ -1,3 +1,688 @@
+    * pkg/lwip: add support for esp32 Ethernet device
+    * pkg/lwip: add IPv4 support for LWIP-stm32
+    * pkg/lwip: start DHCP for a netif with lwip_dhcp
+    * pkg/lwip: add STM32 support
```
I guess here it should be:
```suggestion
    * pkg/lwip: add stm32 ethernet support
```
?

aabadie commented on release-notes.txt:
```diff
@@ -1,3 +1,688 @@
+    * pkg/lwip: start DHCP for a netif with lwip_dhcp
+    * pkg/lwip: add STM32 support
+    * pkg/lwip: enable lwIP for ESP8266
+    + pkg/nanopb: add nanoPb protocol buffers library package
```
```suggestion
    + pkg/nanopb: add Nanopb protocol buffers library package
```

fjmolinas commented:
> Would it be possible to somehow group addition/changes of the same area when there are several. Something like:
> ```
> * sys/arduino:
>   + added Wire (I2C) interface
>   * allow use of arduino libraries...
>   etc
> ```
> 
> ?

@aabadie I did that in the last fixup, what do you think

kb2ma commented on release-notes.txt:
Please remove. gcoap_req_send2() was removed in #12837 for this release,

fjmolinas replied:
True, sorry about that. removed.

kb2ma replied:
No worries. IIRC you should be approaching the "please let this end" phase of release management. :-)

dylad commented on release-notes.txt:
```diff
@@ -1,3 +1,699 @@
+        * uart: implement non-blocking uart write
+        + uart: add hardware flow control support
+        * rtt: enable COUNTSYNC in CTRLA, fixes `rtt_get_counter()` return value
+        * spi: power off spi bus on releas, avoid bus errors affect application code
```
```suggestion
        * spi: power off spi bus on release, avoid bus errors affect application code
```

aabadie commented on release-notes.txt:
```diff
@@ -1,3 +1,699 @@
+
+ - Initial support for MicroPython
+ - Initial support for GNRC based LoRaWAN stack
+ - Initial  (experimental) asynchronous sock support
```
```suggestion
 - Initial (experimental) asynchronous sock support
```

aabadie commented on release-notes.txt:
```diff
@@ -1,3 +1,699 @@
+ - Initial support for MicroPython
+ - Initial support for GNRC based LoRaWAN stack
+ - Initial  (experimental) asynchronous sock support
+ - Extend support for lpc2387 and fe310
```
```suggestion
 - Extend support for lpc2387 and fe310 cpus
```

aabadie commented on release-notes.txt:
```diff
@@ -1,3 +1,699 @@
+ - Extend support for lpc2387 and fe310
+ - Xtimer concurrency/robustness improvement, fixing #8388, #5338 & #5103
+ - Complete re-implementation of esp8266 based on ESP8266 RTOS SDK
+ - Improvements in automatic tests board coverage and test automation
```
```suggestion
 - Improvements in automatic tests
```

aabadie commented on release-notes.txt:
```diff
@@ -1,3 +1,699 @@
+    * boards/nucleo-f207zg: add ethernet configuration
+    * boards/native: allow for native to be reset via SIGUSR1
+    * boards/particle-*: configure PCB antenna for 2.4GHz radio
+    * boards/pic32-clicker: cleanup  uart configuration and initialization, cleanup header 
```
```suggestion
    * boards/pic32-clicker: cleanup uart configuration and initialization, cleanup header 
```

aabadie commented on release-notes.txt:
```diff
@@ -1,3 +1,699 @@
+    * cpu/stm32:
+        * i2c_2: fix read bytes flag
+        - i2c_2: do not support repeated start reading in
+        * stmclk: fix  M-factor shift for SAI PLL
```
```suggestion
        * stmclk: fix M-factor shift for SAI PLL
```

aabadie commented on known_issues.txt:
```diff
@@ -0,0 +1,218 @@
+Known issues
```
It seems that you added this file by mistake ;)

fjmolinas commented:
I pushed fixes to the latest comments and other fixes

- added release statistics
- I forced 72 char line endings
- fixed some white spaces
- fixed formatting

aabadie commented on release-notes.txt:
```diff
@@ -1,3 +1,714 @@
+ - +12 new boards, +7 new drivers, +4 new packages
+
+About 527 pull requests, composed of 1367 commits, have been merged since the
+last release, and about  issues have been solved. 72 people contributed with
```
```suggestion
last release, and about ? issues have been solved. 72 people contributed with
```

fjmolinas replied:
ups

fjmolinas commented:
@RIOT-OS/maintainers please take a last look! I would like to squash and run the CI.

fjmolinas commented:
@aabadie is it ok to squash?

aabadie commented:
>  is it ok to squash?

I'd prefer waiting for other maintainers replies.

fjmolinas commented:
Travis points out a typo in the api, but since this is in code I won't fix it:

```
	release-notes.txt:417: seperator  ==> separator

	release-notes.txt:418: seperator  ==> separator
```

fjmolinas commented on release-notes.txt:
```diff
@@ -1,3 +1,714 @@
+ - +12 new boards, +7 new drivers, +4 new packages
+
+About 527 pull requests, composed of 1367 commits, have been merged since the
+last release, and about 29 issues have been solved. 72 people contributed with
```
I realized I did not filter contributor duplicates

fjmolinas replied:
Addressed and squashed directly.

fjmolinas commented:
@aabadie all green!

aabadie commented:
2 ACKS and all green => go!

fjmolinas commented:
Backport provided in #13272
